docs/source/scheduled-queries.rst

```diff
@@ -0,0 +1,299 @@
+and analyze them automatically. Rules that were not previously possible can be written:
+
+* Detect X failed logins within Y minutes
+* Detect spike in API behavior that is correlated with an increase in # of a different alert/rule
+* Detect elevated API % error rates from specific IP address
+
+
+How do scheduled queries work?
+``````````````````````````````
+This system leverages two main components: AWS Lambda and AWS Step Functions.
+
+First, a CloudWatch scheduled event triggers the execution of a new AWS Step Function State Machine.
+This State Machine manages the lifecycle of Athena queries through the Lambda function. Its sole
+responsibility is to execute the Lambda, wait a predefined window of time, and execute the Lambda again,
+repeating until the Lambda reports it is finished.
+
+The Lambda function is a simple function that starts Athena queries, caches their execution ids, checks
+on their execution status, and uploads results to StreamAlert via Kinesis. Instead of doing all of these
+steps in a blocking fashion and sleeping while it waits for Athena, it runs through all queries in a single
+nonblocking pass, and returns the result of its execution to the State Machine. Once all queries have
+completed and their results sent to StreamAlert, the Lambda returns a "done" flag to the State Machine,
+signalling that this job has been finished.
+
+
+
+Configuration
+-------------
+Scheduled Queries is configured via a single file, ``conf/scheduled_queries.json``.
+
+.. code-block::
+
+  {
+    "enabled": true,
+    "config": {
+      "destination_kinesis": "prefix_prod_streamalert",
+      "sfn_timeout_secs": 3600,
+      "sfn_wait_secs": 30
+    },
+    "packs": {
+      "hourly": {
+        "description": "Runs all hourly queries",
+        "schedule_expression": "rate(1 hour)"
+      },
+      "two_hour": {
+        "description": "Runs all queries run every two hours",
+        "schedule_expression": "rate(2 hours)"
+      },
+      "daily": {
+        "description": "Runs all daily queries",
+        "schedule_expression": "rate(24 hours)"
+      },
+      "two_day": {
+        "description": "Runs all queries that are run once every 2 days",
+        "schedule_expression": "rate(2 days)"
+      }
+    },
+    "lambda_config": {}
+  }
+
+* ``enabled`` — (bool) Pass `true` to activate ScheduledQueries. Leave `false` to disable.
+* ``config.destination_kinesis`` — (str) The name of the Kinesis stream to upload results to.
+* ``config.sfn_timeout_secs`` - (int) Max number of seconds for the state machine to execute.
+* ``config.sfn_wait_secs`` - (int) Time to wait between checks of query status.
+* ``query_packs`` - (dict) The keys of this dict are the **names** of the query packs. This section is discussed in more depth below.
+
+
+Query Packs
+```````````
+Query Packs are batches of scheduled Athena queries that are executed together.
+
+.. code-block::
```

Comment:
add `json` after this please: `.. code-block:: json`

Reply:
Fixed https://github.com/airbnb/streamalert/pull/1128/commits/f35fcbed506b09545158edc15db6573678a8817b